riverraid.py

```diff
@@ -2,70 +2,65 @@
 from numpy import random
 from numpy.core.numeric import ndarray
 from scipy.misc.pilutil import imresize
+from scipy.misc.pilutil import imshow
 from keras.models import Sequential
 from keras.layers import Dense, Dropout, Activation, Flatten
 from keras.layers import Convolution2D, MaxPooling2D
 from keras.optimizers import Adam
+from keras.optimizers import RMSprop
 from keras.utils import np_utils, plot_model
 from keras.datasets import mnist
 from collections import deque
 import random
 import pydot
 import gym
 import scipy.misc
+import os
+import pickle
+import time
 
 # hyperparameters
-C = 10000
-N = 1000000
-NUM_EPISODES = 20000
+NUM_EPISODES = 100000
 NUM_ITERATIONS = 10000
 EPSILON_MIN = 0.1
-ESPILON_DECAY = 0.995
+ESPILON_DECAY = (0.9/1000000)
```

Comment:
I've also played around with reducing the epsilon decay - in reality we may not run a large enough number of episodes for that decay to be representative of the ratio of exploration to optimization that we want